yamux/src/frame/io.rs

```diff
@@ -174,19 +125,18 @@ impl<T: AsyncRead + AsyncWrite + Unpin> Stream for Io<T> {
         let mut this = &mut *self;
         loop {
             log::trace!("{}: read: {:?}", this.id, this.read_state);
-            match this.read_state {
+            let state = mem::replace(&mut this.read_state, ReadState::Init);
```

Comment:
mem::replace will leave read_state in ReadState::Init state if we exit the function with one of the two ready! macro below.

Reply:
Good catch!